frontend/src/metabase-lib/Question.ts

```diff
@@ -567,22 +560,17 @@ class Question {
     });
   }
 
-  syncColumnsAndSettings(previousQuestion?: Question, queryResults?: Dataset) {
+  syncColumnsAndSettings(queryResults?: Dataset, prevQueryResults?: Dataset) {
     let question = this;
-    const query = question.query();
-    const { isNative } = Lib.queryDisplayInfo(query);
 
     if (queryResults && !queryResults.error) {
       question = question._syncTableColumnSettings(queryResults);
-    }
 
-    if (previousQuestion) {
-      const previousQuery = previousQuestion.query();
-      const { isNative: isPreviousQuestionNative } =
-        Lib.queryDisplayInfo(previousQuery);
-
-      if (!isNative && !isPreviousQuestionNative) {
```

Comment:
We now check column source to ignore native queries so we don't depend on the query in any way.